common/flatpak-utils.c

```diff
@@ -3122,6 +3122,29 @@ flatpak_complete_word (FlatpakCompletion *completion,
   g_print ("%s\n", rest);
 }
 
+static gboolean
+switch_already_in_line (FlatpakCompletion *completion,
+                        GOptionEntry      *entry)
+{
+  guint i = 0;
+  guint line_part_len = 0;
+  g_auto(GStrv) line = NULL;
+
+  line = g_strsplit (completion->line, " ", -1);
```

Comment:
Look at completion->argv[] instead, which is pre-split (and shell-unescaped, etc)